assets/wizards/newspack/views/settings/connections/custom-events.tsx

```diff
@@ -0,0 +1,143 @@
+						'newspack-plugin'
+					),
+					'ga4_invalid_measurement_id'
+				)
+			);
+			return;
+		}
+		if ( ! ga4Credentials.measurement_protocol_secret ) {
+			setError(
+				new WizardError(
+					__(
+						'You need a valid Measurement API Secret to activate Newspack Custom Events.',
+						'newspack-plugin'
+					),
+					'ga4_invalid_measurement_protocol_secret'
+				)
+			);
+			return;
+		}
+		resetError();
+	}, [ ga4Credentials.measurement_id, ga4Credentials.measurement_protocol_secret ] );
+
+	function updateGa4Credentials() {
+		wizardApiFetch< Ga4Credentials >(
+			{
+				path: '/newspack/v2/wizard/analytics/ga4-credentials',
+				method: 'POST',
+				data: {
+					measurement_id: ga4Credentials.measurement_id,
+					measurement_protocol_secret: ga4Credentials.measurement_protocol_secret,
+				},
+			},
+			{
+				onSuccess( fetchedData ) {
+					window.newspackSettings.connections.sections.customEvents = {
+						...window.newspackSettings.connections.sections.customEvents,
+						...fetchedData,
+					};
+					setGa4Credentials( fetchedData );
+				},
+			}
+		);
+	}
+
+	return (
+		<div className="newspack__analytics-configuration">
+			<div className="newspack__analytics-configuration__header">
+				<p>
+					{ __(
+						"Newspack already sends some custom event data to your GA account, but adding the credentials below enables enhanced events that are fired from your site's backend. For example, when a donation is confirmed or when a user successfully subscribes to a newsletter.",
+						'newspack-plugin'
+					) }
+				</p>
+			</div>
+
+			{ errorMessage && <Notice isError noticeText={ errorMessage } /> }
+			<Grid noMargin rowGap={ 16 }>
+				<TextControl
+					value={ ga4Credentials.measurement_id }
+					label={ __( 'Measurement ID', 'newspack-plugin' ) }
+					help={ __(
+						'You can find this in Site Kit Settings, or in Google Analytics > Admin > Data Streams and clickng the data stream. Example: G-ABCDE12345',
+						'newspack-plugin'
+					) }
+					onChange={ ( value: string ) =>
+						setGa4Credentials( { ...ga4Credentials, measurement_id: value } )
+					}
+					autoComplete="off"
+				/>
+				<TextControl
+					type="password"
+					value={ ga4Credentials.measurement_protocol_secret }
+					label={ __( 'Measurement Protocol API Secret', 'newspack-plugin' ) }
+					help={ __(
+						'Generate an API secret from your GA dashboard in Admin > Data Streams and opening your data stream. Select "Measurement Protocol API secrets" under the Events section. Create a new secret.',
+						'newspack-plugin'
+					) }
+					onChange={ ( value: string ) =>
+						setGa4Credentials( { ...ga4Credentials, measurement_protocol_secret: value } )
+					}
+					autoComplete="one-time-code"
+				/>
+			</Grid>
+			<Button
+				className="newspack__analytics-newspack-custom-events__save-button"
+				variant="primary"
+				onClick={ updateGa4Credentials }
+				disabled={ !! errorMessage }
```

Comment:
There is some circular logic here when the fields start off empty (when a publisher wants to enter credentials for the first time). The error message will appear by default, and so the save button is disabled. However the error message hook will only reset the error message when the credentials state is updated, which can only be done by clicking the save button.

Reply:
Related to this, there is currently no way to remove credentials once they are entered. We should either allow saving empty fields, or add a reset button (preferably a reset button)

Reply:
I like the `reset` suggestion. Have added:
* Registered endpoint
* Endpoint callback
* UI to facilitate reseting inputs

Reply:
Thanks! Working great now 👍